File: Orion/Controllers/Requests/Relation/AddRelationRequest.cs
using FluentValidation;

namespace Orion.Controllers.Requests.Relation
{
    public class AddRelationRequest
    {
        public string Name { get; set; }
        public string? Code { get; set; }
        public string? VatNumber { get; set; }
        public string? Website { get; set; }
        public string? Street { get; set; }
        public string? PostalCode { get; set; }
        public string? City { get; set; }
        public string? Country { get; set; }
        public string? Language { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? MobilePhone { get; set; }
        public bool IsCompany { get; set; }
    }

    public class AddRelationRequestValidator : AbstractValidator<AddRelationRequest>
    {
        public AddRelationRequestValidator()
        {
            RuleFor(req => req.Name).NotEmpty().MaximumLength(120);
            RuleFor(req => req.Code).MaximumLength(120);
            RuleFor(req => req.VatNumber).MaximumLength(25);
            RuleFor(req => req.Website).MaximumLength(2048);
            RuleFor(req => req.Street).MaximumLength(95);
            RuleFor(req => req.PostalCode).MaximumLength(10);
            RuleFor(req => req.City).MaximumLength(35);
            RuleFor(req => req.Country).MaximumLength(50);
            RuleFor(req => req.Language).MaximumLength(10);
            RuleFor(req => req.Email).MaximumLength(62);
            RuleFor(req => req.Phone).MaximumLength(30);
            RuleFor(req => req.MobilePhone).MaximumLength(30);
            RuleFor(req => req.IsCompany).NotEmpty();
        }
    }

}

File: Orion/ClientApp/src/app/shared/requests/relation/add-relation.request.ts
export class AddRelationRequest {
  name: string;
  code?: string | null;
  vatNumber?: string | null;
  website?: string | null;
  street?: string | null;
  postalCode?: string | null;
  city?: string | null;
  country?: string | null;
  language?: string | null;
  email?: string | null;
  phone?: string | null;
  mobilePhone?: string | null;
  isCompany: boolean;

  constructor(
    name: string,
    isCompany: boolean,
    code?: string | null,
    vatNumber?: string | null,
    website?: string | null,
    street?: string | null,
    postalCode?: string | null,
    city?: string | null,
    country?: string | null,
    language?: string | null,
    email?: string | null,
    phone?: string | null,
    mobilePhone?: string | null
  ) {
    this.name = name;
    this.isCompany = isCompany;
    this.code = code;
    this.vatNumber = vatNumber;
    this.website = website;
    this.street = street;
    this.postalCode = postalCode;
    this.city = city;
    this.country = country;
    this.language = language;
    this.email = email;
    this.phone = phone;
    this.mobilePhone = mobilePhone;
  }
}
